src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -532,108 +535,55 @@ private Rect ApplyMarginAndAlignments(View view, Rect frame)
 			);
 		}
 
-		/// <summary>
-		/// Get actual width based on MinWidth, MaxWidth, Width and HorizontalAlignment
-		/// </summary>
-		private double GetActualWidth(double width,
-			HorizontalAlignment childHorizontalAlignment,
-			double childMaxWidth,
-			double childMinWidth,
-			double childWidth,
-			Thickness childMargin,
-			bool hasChildWidth,
-			bool hasChildMaxWidth,
-			bool hasChildMinWidth,
-			Size desiredSize)
-		{
-			//Default value
-			//childHorizontalAlignment != HorizontalAlignment.Stretch
-			var actualWidth = Min(width, desiredSize.Width);
-
-			if (hasChildWidth)
-			{
-				actualWidth = Min(childWidth + childMargin.Left + childMargin.Right, width);
-			}
-			else if (hasChildMaxWidth && hasChildMinWidth)
-			{
-				actualWidth = Min(childMaxWidth + childMargin.Left + childMargin.Right,
-						childHorizontalAlignment == HorizontalAlignment.Stretch
-						? width
-						: desiredSize.Width
-					);
-
-				actualWidth = Max(childMinWidth + childMargin.Left + childMargin.Right, actualWidth);
-			}
-			else if (hasChildMaxWidth)
-			{
-				actualWidth = Min(childMaxWidth + childMargin.Left + childMargin.Right,
-						childHorizontalAlignment == HorizontalAlignment.Stretch
-						? width
-						: desiredSize.Width
-					);
-			}
-			else if (hasChildMinWidth)
-			{
-				actualWidth = Max(childMinWidth + childMargin.Left + childMargin.Right,
-						childHorizontalAlignment == HorizontalAlignment.Stretch
-						? width
-						: desiredSize.Width
-					);
-			}
-
-			return actualWidth;
-		}
-
-		/// <summary>
-		/// Get actual height based on MinHeight, MaxHeight, Height and VerticalAlignment
-		/// </summary>
-		private double GetActualHeight(double height,
-			VerticalAlignment childVerticalAlignment,
+		private double GetActualSize(
+			double size,
+			bool isStretch,
 			double childMaxHeight,
 			double childMinHeight,
 			double childHeight,
-			Thickness childMargin,
+			double childMargins,
 			bool hasChildHeight,
 			bool hasChildMaxHeight,
 			bool hasChildMinHeight,
-			Size desiredSize)
+			double desiredSize,
+			double frameSize)
 		{
 			//Default value
 			//childVerticalAlignment != VerticalAlignment.Stretch
-			var actualHeight = Min(height, desiredSize.Height);
+			var actualHeight = Min(size, desiredSize);
 
 			if (hasChildHeight)
 			{
-				actualHeight = Min(childHeight + childMargin.Top + childMargin.Bottom, height);
+				actualHeight = Min(childHeight + childMargins, size);
```

Comment:
All references to Height should be replaced with Size

Reply:
Yep you're right!  It's already merged, so a new PR is required for that...